Seems like depending on Python version things are implemented differently:
```
c:\projects\micropython\tests\basics>c:\Python34\python.exe -BS .\containment.py

1 in {1: 2}: True
1 not in {1: 2}: False
Traceback (most recent call last):
  File ".\containment.py", line 4, in <module>
    print("{} in {}: {}".format(i, o, i in o))
TypeError: Type dict_keys doesn't define __format__
```
From Python 3.5 on this is fixed. Imo the change of using `-BS` is worth it, but that does mean docs/READMES need to explicitly state that it's needed (at least for the test suite) or, maybe easier, fix this particular test.

--- a/tests/basics/containment.py
+++ b/tests/basics/containment.py
@@ -1,8 +1,8 @@
 # sets, see set_containment
 for i in 1, 2:
     for o in {1:2}, {1:2}.keys():
-        print("{} in {}: {}".format(i, o, i in o))
-        print("{} not in {}: {}".format(i, o, i not in o))
+        print("{} in {}: {}".format(i, str(o), i in o))
+        print("{} not in {}: {}".format(i, str(o), i not in o))
 
 haystack = "supercalifragilistc"
 for needle in [haystack[i:] for i in range(len(haystack))]:

--- a/tests/run-tests
+++ b/tests/run-tests
@@ -26,6 +26,10 @@ else:
     CPYTHON3 = os.getenv('MICROPY_CPYTHON3', 'python3')
     MICROPYTHON = os.getenv('MICROPY_MICROPYTHON', base_path('../ports/unix/micropython'))
 
+# Use CPython options to not save .pyc files, to only access the core standard library
+# (not site packages which may clash with u-module names), and improve start up time.
+CPYTHON3_CMD = [CPYTHON3, "-BS"]
+
 # mpy-cross is only needed if --via-mpy command-line arg is passed
 MPYCROSS = os.getenv('MICROPY_MPYCROSS', base_path('../mpy-cross/mpy-cross'))
 
@@ -319,7 +323,7 @@ def run_tests(pyb, tests, args, result_dir):
             upy_float_precision = int(upy_float_precision)
         has_complex = run_feature_check(pyb, args, base_path, 'complex.py') == b'complex\n'
         has_coverage = run_feature_check(pyb, args, base_path, 'coverage.py') == b'coverage\n'
-        cpy_byteorder = subprocess.check_output([CPYTHON3, base_path('feature_check/byteorder.py')])
+        cpy_byteorder = subprocess.check_output(CPYTHON3_CMD + [base_path('feature_check/byteorder.py')])
         skip_endian = (upy_byteorder != cpy_byteorder)
 
     # These tests don't test slice explicitly but rather use it to perform the test
@@ -498,7 +502,7 @@ def run_tests(pyb, tests, args, result_dir):
         else:
             # run CPython to work out expected output
             try:
-                output_expected = subprocess.check_output([CPYTHON3, '-B', test_file])
+                output_expected = subprocess.check_output(CPYTHON3_CMD + [test_file])
                 if args.write_exp:
                     with open(test_file_expected, 'wb') as f:
                         f.write(output_expected)

